### ISSUE TITLE
`Package scrhack Warning: unknown \newfloat definition found!` (KOMA-script)

### ISSUE THREAD

**Issue by goretkin** [fmarotta/kaobook#148]
### Prerequisites

* [ X ] Put an X between the brackets on this line if you have done all of the following:
    * Check that you are using the latest version of kaobook.
    * Check that your issue isn't already filed: https://github.com/fmarotta/kaobook/issues

### Description
This might be specific to KOMA-script, but I am unfamiliar with it, and I'm encountering it through kaobook.

running `latexmk -pdf  mwe.tex` with the MWE below succeeds, but there is a warning:
```
[...]
(/usr/local/texlive/2021/texmf-dist/tex/latex/koma-script/scrhack.sty
(/usr/local/texlive/2021/texmf-dist/tex/latex/koma-script/float.hak

Package scrhack Warning: unknown \newfloat definition found!
(scrhack)                Maybe you are using a unsupported float version.

) (/usr/local/texlive/2021/texmf-dist/tex/latex/koma-script/floatrow.hak)
(/usr/local/texlive/2021/texmf-dist/tex/latex/koma-script/listings.hak)
[...]
```

### Minimal Working Example
[You can use the following document to help you create a minimal working example.]
```latex
\documentclass{kaobook}

\usepackage{mwe}

\begin{document}
\mainmatter

\begin{figure}
	\centering
	\includegraphics[width=0.5\columnwidth]{example-image}
	\caption{
		A figure caption
	}
\end{figure}


\end{document}
```

### Additional Information

```
$ pdflatex --version
pdfTeX 3.141592653-2.6-1.40.22 (TeX Live 2021)
kpathsea version 6.3.3
Copyright 2021 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.37; using libpng 1.6.37
Compiled with zlib 1.2.11; using zlib 1.2.11
Compiled with xpdf version 4.03
```

**Comment by fmarotta** [comment]
It doesn't seem to be kao-specific, as it occurs even with this document, which uses `scrbook`:
```
\documentclass{scrbook}

\usepackage{floatrow}
\usepackage{scrhack}

\begin{document}

\end{document}
```

It is enough to load `floatrow` and `scrhack` at the same time to trigger the warning. `scrhack` tries to make `floatrow` compatible with KOMAScript, but it supports only some versions of the latter. Probably it supports only older versions of `floatrow`, so one possibility would be to downgrade `floatrow`.

**Comment by goretkin** [comment]
Thanks! I opened an issue here: https://sourceforge.net/p/koma-script/tickets/9/

**Comment by komascript** [comment]
I seems that `kaobook` loads `scrhack` after `floatrow`. It is recommended to load it before loading any of the packages, that should be patched.

Nevertheless, loading `float.hak` if `floatrow` has already been loaded is a bug in `scrhack` that will be fixed in KOMA-Script 3.34.

**Comment by fmarotta** [comment]
Thank you! I was confused by [this post on TeX.SE](https://tex.stackexchange.com/questions/156240/latex-packages-minted-and-scrhack). Now `scrhack` is loaded before any of the patched packages.
